File: 프로그래머스/lv0/120833. 배열 자르기/배열 자르기.java
import java.util.*;

class Solution {
    public int[] solution(int[] numbers, int num1, int num2) {
        int[] answer = new int[numbers.length];
        answer = Arrays.copyOfRange(numbers, num1, num2 + 1);        
        return answer;
    }
}



// class Solution {
//     public int[] solution(int[] numbers, int num1, int num2) {
//         int[] answer = new int[num2 - num1 + 1];
        
//         for (int i = num1, j = 0; i <= num2; i++, j++) {
//             answer[j] = numbers[i]; 
//             // System.out.println("j : " + j);
//             // System.out.println("numbers[i] : " + numbers[i]);
//         }   
        
//         return answer;
//     }
// }

File: 프로그래머스/lv0/120833. 배열 자르기/배열 자르기.js
function solution(numbers, num1, num2) { 
    return numbers.slice(num1, num2+1);
}


// function solution(numbers, num1, num2) {
//     let answer = [];
    
//     for (let i = num1; i <= num2; i++) {
//         answer.push(numbers[i]);        
//         console.log("i : " + i);
//         console.log("numbers[i] : " + numbers[i]);
//     }    
    
//     return answer;
// }